src/views/transaction/transaction.view.jsx

```diff
@@ -139,104 +142,74 @@ function Transaction ({
                         {getTransactionTypeLabel(transactionTask.data.type)}
                       </Col>
                     </Row>
-                    {transactionTask.data.fromHezEthereumAddress
+                    {transactionTask.data.fromHezEthereumAddress && transactionTask.data.fromHezEthereumAddress !== INTERNAL_ACCOUNT_ETH_ADDR
                       ? (
                         <Row>
                           <Col>From</Col>
                           <Col link>
                             <Row wrapped>
-                              <CopyToClipboardButton content={transactionTask.data.fromAccountIndex} />
+                              <CopyToClipboardButton content={transactionTask.data.fromHezEthereumAddress} />
                               <Col wrapped>
                                 <Link to={`/user-account/${transactionTask.data.fromHezEthereumAddress}`}>
-                                  {transactionTask.data.fromAccountIndex}
+                                  {transactionTask.data.fromHezEthereumAddress}
                                 </Link>
                               </Col>
                             </Row>
                           </Col>
                         </Row>
                         )
                       : <></>}
-                    {!transactionTask.data.fromHezEthereumAddress && transactionTask.data.fromBjj
+                    {(!transactionTask.data.fromHezEthereumAddress || transactionTask.data.fromHezEthereumAddress === INTERNAL_ACCOUNT_ETH_ADDR) && (transactionTask.data.fromBjj || transactionTask.data.fromBJJ)
                       ? (
                         <Row>
                           <Col>From</Col>
                           <Col link>
                             <Row wrapped>
-                              <CopyToClipboardButton content={transactionTask.data.fromAccountIndex} />
+                              <CopyToClipboardButton content={transactionTask.data.fromBjj ? transactionTask.data.fromBjj : transactionTask.data.fromBJJ} />
```

Comment:
can be written as `transactionTask.data.fromBjj || transactionTask.data.fromBJJ`